client/network/src/service/metrics.rs

```diff
@@ -75,10 +75,9 @@ pub struct Metrics {
 	pub pending_connections: Gauge<U64>,
 	pub pending_connections_errors_total: CounterVec<U64>,
 	pub requests_in_failure_total: CounterVec<U64>,
-	pub requests_in_success_total: HistogramVec,
+	pub requests_in_duration_building_response: HistogramVec,
 	pub requests_out_failure_total: CounterVec<U64>,
 	pub requests_out_success_total: HistogramVec,
-	pub requests_out_started_total: CounterVec<U64>,
```

Comment:
This metric was there to potentially detect requests that are stuck, or still in progress in general.
What's the reason for removing it?

Reply:
I have to admit that I dropped it with removing block_requests.rs but never reintroduced it when switching to request-responses.rs.

Without the metric one can not tell how many requests are in-flight at a given point in time, though one can estimate it via `requests_out_success_total`. Given that `request_responses.rs` does not spawn new connections, requests to the same peer are done in sequence, capped with a limit and refused requests (requests over the limit) are reported in `requests_out_failure_total` I am not sure we still need it.

It does not take a lot of work to reimplement it, nor to maintain it, thus I am fine reintroducing the metric. Let me know what you prefer.

Reply:
I don't really care one way or the other. The metric is only useful in the presence of a bug, so it's a matter of being optimistic or pessimistic.